### PR TITLE
bug fix: loss_mask should not be shifted. only labels needs shifting

### PR DESCRIPTION
**Pull request by yeyu-nvidia**
## What does this PR do?

**Type of change:** ? Bug fix

**Overview:** 
loss mask should not be shifted. Only labels need to be shifted by 1 token.

## Usage

```python
# Add a code snippet demonstrating how to use this
```

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected label alignment in speculative decoding examples, preventing unintended shifting of the loss mask and loss on incorrect positions.
  * Ensures loss is computed on the intended tokens, improving training stability, convergence, and metric reliability.
  * Reduces accidental inclusion of padding/shifted tokens in loss calculations.
  * No changes to public APIs or user-facing configuration.

### CODE REVIEW

File: examples/speculative_decoding/eagle_utils.py [hunk 1]
```diff
@@ -101,8 +101,7 @@ def get_role_content(item):
                 labels[indices] = input_ids[indices]
                 loss_mask[indices] = 1
 
-        # Shift loss_mask and labels to the left by 1 token
-        loss_mask = torch.cat([loss_mask[1:], torch.zeros(1, dtype=loss_mask.dtype)])
+        # Shift labels to the left by 1 token
```
Comment (h-guo18) [inline examples/speculative_decoding/eagle_utils.py:104]: I think this shift is needed, since we are predicting the +1 token.
e.g. Input_ids= [a, b, C, D, E,] where C, D, E is the answer, then b is the first position we want to count loss, since it's used to predict C. Then the loss mask should be [0, 1, 1, 1, 0].
